### ISSUE TITLE
Refactor the test api and e2e test script to avoid starting unnecessary services

### ISSUE THREAD

**Issue by whynowy** [numaproj/numaflow#1726]
Latency should be minimal since it is only introduced during the testing of one component, and each component is run in parallel. I'll give it some thought and create a separate PR if it can be avoided.

_Originally posted by @yhl25 in https://github.com/numaproj/numaflow/pull/1725#discussion_r1594961795_

**Comment by samhith-kakarla** [comment]
i will try and look into this

**Comment by whynowy** [comment]
Closed by https://github.com/numaproj/numaflow/pull/1760.
